executor/tests/witgen.rs

```diff
@@ -0,0 +1,36 @@
+use powdr_number::{FieldElement, GoldilocksField};
+use powdr_pipeline::pipeline::Columns;
+use powdr_pipeline::Pipeline;
+use std::sync::Arc;
+
+fn run_witgen_pil<T: FieldElement>(pil: &str) -> Arc<Columns<T>> {
+    env_logger::init();
+
+    Pipeline::default()
+        .from_pil_string(pil.to_string())
+        .compute_witness()
+        .unwrap()
+}
+
+#[test]
+#[should_panic = "Publics are referenced by more than one machine: {\"public\"}"]
+fn two_machines_conflicting_public() {
```

Comment:
I see that this test is supposed to test witness generation, but shouldn't we just put it into the pipeline crate if it's not a unit test? That way we avoid the dependency.

Reply:
Can you elaborate which workflow would be faster then? I think it does fit will in the `executor` crate...

Reply:
My point is that a unit test should only depend on the crate it tests. If it's not a unit test but an integration test, it can depend on whatever else. But in that case, the test should be in the crate that also depends on whatever else and that's the pipeline crate.

The issue here is that when we build the binary for the executor and it depends on the pipeline, we are linking all the other stuff including weird field arithmetic crates and wasm stuff pulled in through plonky3. Of course it is removed before the final binary is created (at least I assume), but not sure at which stage that happens and it needs to be detected first.

Reply:
What about writing `run_witgen_pil` in a way which does not rely on pipeline? Would it be acceptable to have the compiler as a dev dependency?

Reply:
Or if pipeline is the easiest UX, have a `light` feature flag on pipeline which does not include proving? Basically each crate could dev-depend on pipeline with everything turned on before itself.

Reply:
I don't think that a feature flag helps here. As far as I understand feature flags, they are always additive, in the sense that if one compilation part sets a feature flag, it is set for all other compilation parts in the compilation run.

Reply:
Oh and about adding the compiler as a dev dependency: The parser is a gigantic dependency and takes so long to build. It's a nightmare if you change anything in the grammar, it causes all the stuff that builds on the parser to be re-built.

Reply:
If you built only this crate, wouldn't is use only the flag? That was my assumption here. If building the workspace it would be additive but it makes sense because you'd need the whole thing for other crates anyway?

Re not relying on the compiler: then it means that we cannot write tests based on source code if we can't parse it? Or is your point that any test which requires this should be in pipeline?

Reply:
I moved the test to the `pipeline` crate!